lib/src/native/realm_core.dart

```diff
@@ -1566,3 +1783,36 @@ enum _HttpMethod {
   put,
   delete,
 }
+
+extension on realm_timestamp_t {
+  DateTime toDart() {
+    return DateTime.fromMicrosecondsSinceEpoch(seconds * 1000000 + nanoseconds ~/ 1000, isUtc: true);
+  }
+}
+
+extension on realm_string_t {
+  String? toDart() {
+    if (data == nullptr) return null;
+    return data.cast<Utf8>().toDartString();
+  }
+}
+
+extension on ObjectId {
+  Pointer<realm_object_id> toCapi(Allocator allocator) {
+    final result = allocator<realm_object_id>();
+    for (var i = 0; i < 12; i++) {
+      result.ref.bytes[i] = bytes[i];
+    }
+    return result;
+  }
+}
+
+extension on realm_object_id {
+  ObjectId toDart() {
```

Comment:
I would prefer we had `toDart` extensions methods on `NativeType` subclasses, and `toCapi` extensions on the dart types we need to convert. I think it will read much nicer. I don't see much benefit in adding the target type in the name if avoidable, since it is obvious from context, and easily discoverable if you are in doubt.